fem/fespace.cpp

```diff
@@ -93,16 +92,68 @@ FiniteElementSpace::FiniteElementSpace(const FiniteElementSpace &orig,
    Constructor(mesh, NURBSext, fec, orig.vdim, orig.ordering);
 }
 
-int FiniteElementSpace::GetOrder(int i) const
+void FiniteElementSpace::SetElementOrder(int i, int p)
 {
-   Geometry::Type GeomType = mesh->GetElementBaseGeometry(i);
-   return fec->FiniteElementForGeometry(GeomType)->GetOrder();
+   MFEM_VERIFY(sequence == mesh->GetSequence(), "space has not been Updated()");
+   MFEM_VERIFY(i >= 0 && i < GetNE(), "invalid element index");
+   MFEM_VERIFY(p >= 0 && p <= MaxVarOrder, "order ouf of range");
+   MFEM_ASSERT(!elem_order.Size() || elem_order.Size() == GetNE(),
```

Comment:
Using `!IsVariableOrder()` would make all the `!elem_order.Size()` more self-documented.

Reply:
I also don't understand why there is this `MFEM_ASSERT`, `!elem_order.Size()` is treated just below.

Reply:
Also, isn't it expected that `elem_order.Size() == GetNE()`?

Reply:
Yes it is expected, the expression `!elem_order.Size() || elem_order.Size() == GetNE()` is an invariant that needs to hold, and we check it in debug with an ASSERT, isn't that what asserts are for?

Reply:
To break the expression down, the array `elem_order` is either empty (regular constant-order space) or it is exactly of length `GetNE()` (in case of variable-order space) -- nothing else, which is exactly what the ASSERT checks.

Reply:
As for using `IsVariableOrder`, I tend to use it everywhere else, but this particular method is one of the few whose very job is to manipulate `elem_order` directly so it makes sense IMO that it's using `elem_order.Size()` directly.